examples/tools/bevymark.rs

```diff
@@ -209,19 +222,19 @@ fn spawn_birds(
 }
 
 fn movement_system(time: Res<Time>, mut bird_query: Query<(&mut Bird, &mut Transform)>) {
-    for (mut bird, mut transform) in bird_query.iter_mut() {
+    bird_query.for_each_mut(|(mut bird, mut transform)| {
```

Comment:
I think that using the for_each pattern here is worth it, but we should explain to beginners that this is faster (and ideally link to an explanation why).

Reply:
Agreed, I'd actually love to know why this is faster. [This PR](https://github.com/bevyengine/bevy/pull/1525#:~:text=Uber%20Fast%20%22for_each%22%20Query%20Iterators) and said it was faster but not why.

Reply:
Added a note, though I have no idea if it's correct. :sweat_smile:

Reply:
I can't comment on the correctness of that note, but I feel like something like
```
// `for_each_mut` is faster than `.iter`, but can't be chained like a normal iterator.
```
would be enough